src/MPI.jl

```diff
@@ -160,8 +161,15 @@ using PrecompileTools: @compile_workload
     @compile_workload begin
         # Running the load time hooks here shaves off a significant amount of loading time,
         # see also https://github.com/JuliaParallel/MPI.jl/pull/728
-        foreach(hook -> hook(), _mpi_load_time_hooks)
+        foreach(call, _mpi_load_time_hooks)
     end
+    # We insert some explicit precompile statements here. The corresponding methods
+    # are likely to be called from anyone using MPI in Julia. Thus, it is reasonable
+    # to compile them ahead of time instead of compiling them in parallel for each
+    # run.
+    Base.precompile(Tuple{typeof(Init)})
```

Comment:
Are you sure the argument is correct? Shouldn't it be 
```suggestion
    Base.precompile(Init, ())
```
and similarly below.

Reply:
That's the output I get from SnoopCompile.jl, e.g., `Base.precompile(Tuple{typeof(main)})` in https://timholy.github.io/SnoopCompile.jl/stable/snoopi_deep_parcel/#SnoopCompile.write

Reply:
Well, that's not what documentation says: https://docs.julialang.org/en/v1/base/base/#Base.precompile

Reply:
Updated in c5393d51eeaaae507b8a30cc62b8b67340b73be6

Reply:
Maybe the docs in SnoopCompile and/or the generated code are out of date? The method referenced by @giordano internally calls the variant used by @ranocha, afaict:
https://github.com/JuliaLang/julia/blob/8e630552924eac54c809aa7bc30871c7df1582d3/base/loading.jl#L2871

Reply:
That's something to be discussed at the SnoopCompile.jl repo. The current version of this PR should be fine here, isn't it?